lib/saluki-env/src/autodiscovery/providers/remote_agent.rs

```diff
@@ -0,0 +1,84 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+use futures::StreamExt;
+use tokio::sync::broadcast::{self, Receiver, Sender};
+use tokio::sync::OnceCell;
+use tracing::{debug, info, warn};
+
+use crate::autodiscovery::{AutodiscoveryEvent, AutodiscoveryProvider, Config};
+use crate::helpers::remote_agent::RemoteAgentClient;
+
+/// An autodiscovery provider that uses the Datadog Agent's internal gRPC API to receive autodiscovery updates.
+pub struct RemoteAgentAutoDiscoveryProvider {
+    client: RemoteAgentClient,
+    sender: Sender<AutodiscoveryEvent>,
+    listener_init: OnceCell<()>,
+}
+
+impl RemoteAgentAutoDiscoveryProvider {
+    /// Creates a new `RemoteAgentAutoDiscoveryProvider` that uses the remote client to receive autodiscovery updates.
+    pub fn new(client: RemoteAgentClient) -> Self {
+        let (sender, _) = broadcast::channel::<AutodiscoveryEvent>(100);
+
+        Self {
+            client,
+            sender,
+            listener_init: OnceCell::new(),
+        }
+    }
+
+    async fn start_background_listener(&self) {
+        debug!("Starting autodiscovery background listener");
```

Comment:
General comment: we should always strive to use proper punctuation in log messages, so just adding a period to the end of this.